app/view/sdl/ServiceUpdatePopUp.js

```diff
@@ -0,0 +1,184 @@
+  {
+    elementId: 'ServiceUpdatePopUp',
+    classNames: 'ServiceUpdatePopUp',
+    classNameBindings: [
+      'active'
+    ],
+    childViews: [
+      'message',
+      'progressIndicatorView',
+      'backButton'
+    ],
+
+    timer: null,
+    active: false,
+    content: '',
+    progress: false,
+
+    progressIndicatorView: Em.View.extend(
+      {
+        elementId: 'progress',
+        classNameBindings: 'this.parentView.progress:progress'
+      }
+    ),
+    backButton: SDL.Button.extend(
+      {
+        classNames: 'button backButton',
+        text: 'X',
+        click: function() {
+          this._parentView.deactivate();
+        },
+        buttonAction: true,
+        onDown: false
+      }
+    ),
+    message: SDL.Label.extend(
+      {
+        elementId: 'text',
+        classNames: 'text',
+        contentBinding: 'parentView.content'
+      }
+    ),
+
+    /**
+     * @function deactivate
+     * @desc Deactivate ServiceUpdatePopUp
+     * 
+     */
+    deactivate: function() {
+      this.set('active', false);
+      this.set('progress', false);
+      this.set('content', '');
+    },
+
+    /**
+     * @function activate
+     * @desc activate ServiceUpdatePopUp
+     * @param {string} serviceType 
+     * @param {string} serviceEvent 
+     * @param {string} serviceReason 
+     */
+    activate: function(serviceType, serviceEvent, serviceReason) {
+      if(serviceType === undefined) {
+        return;
+      }
+
+      switch (serviceEvent) {
+        case 'REQUEST_RECEIVED': {
+          if(serviceType == 'VIDEO') {
+            this.set('content', 'Starting Video Stream');
+          }
+          else if(serviceType == 'AUDIO') {
+            this.set('content', 'Starting Audio Stream');
+          }
+          else {
+            return;
+          }
+          this.set('progress', true);
+          this.set('active', true);
+          break;
+        }
+        case 'REQUEST_ACCEPTED': {
+          this.deactivate();
+          break;
+        }
+        case 'REQUEST_REJECTED': {
+          this.rejectedRequest(serviceReason);
+          var self = this;
+          this.timer = setTimeout(
+            function() {
+              self.deactivate();
+            },
+            5000
+          );
+        }
+        default: {
+          if(serviceReason !== undefined) {
+            this.rejectedRequest(serviceReason);
+            var self = this;
+            this.timer = setTimeout(
+              function() {
+                self.deactivate();
+              },
+              5000
+            );
+            break;
+          }
+          break;
+        }
+      }
+    },
+
+    /**
+     * @function rejectedRequest
+     * @desc Processing reason of rejected request
+     * @param {string} reason 
+     */
+    rejectedRequest: function(reason) {
+      switch(reason) {
+        case 'PTU_FAILED': {
+          this.set('active', true);
+          this.set('progress',false);
```

Comment:
@ValeriiMalkov `this.set('active', true);` and `this.set('progress',false)`; the same in all case, please move this statement before `switch`.

Reply:
@BSolonenko Fixed in commit 9617276